internal/mongo/election_period.go

```diff
@@ -0,0 +1,121 @@
+package mongo
+
+import (
+	"context"
+	"time"
+
+	"github.com/Coflnet/sky-mayor/internal/model"
+	"github.com/rs/zerolog/log"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func InsertElectionPeriod(electionPeriod *model.ElectionPeriod) error {
+	ctx, _ := context.WithTimeout(context.Background(), time.Second*30)
+
+	electionPeriod.ID = primitive.NewObjectID()
+
+	result, err := electionPeriodCollection.InsertOne(ctx, electionPeriod)
+	if err != nil {
+		log.Error().Err(err).Msgf("%v id, there was an error when trying to insert election period with id %d", electionPeriod.ID, electionPeriod.Year)
+		return err
+	}
+
+	log.Info().Msgf("%v id, election period %d inserted", result.InsertedID, electionPeriod.Year)
+	return nil
+}
+
+func InsertElectionPeriods(electionPeriods []*model.ElectionPeriod) error {
+	ctx, _ := context.WithTimeout(context.Background(), time.Second*30)
+
+	var insertList []interface{}
+	for i := 0; i < len(electionPeriods); i++ {
+		electionPeriods[i].ID = primitive.NewObjectID()
+		insertList = append(insertList, electionPeriods[i])
+	}
+
+	result, err := electionPeriodCollection.InsertMany(ctx, insertList)
+	if err != nil {
+		log.Error().Err(err).Msgf("There was an error when trying to insert multiple election periods. Inserts: %v", electionPeriods)
+		return err
+	}
+
+	log.Info().Msgf("inserted multple election periods: ", result.InsertedIDs)
+	return nil
+}
+
+func UpdateElectionPeriod(electionPeriod *model.ElectionPeriod) error {
+	ctx, _ := context.WithTimeout(context.Background(), time.Second*30)
+
+	_, err := electionPeriodCollection.ReplaceOne(ctx, bson.D{{"_id", electionPeriod.ID}}, electionPeriod)
+
+	if err != nil {
+		log.Error().Err(err).Msgf("there was an error when updating eleciton period with id %s", electionPeriod.ID)
+		return err
+	}
+
+	log.Info().Msgf("id: %v, successfully replaced election period of year %d", electionPeriod.ID, electionPeriod.Year)
+	return nil
+}
+
+func GetElectionPeriodByYear(year int) (*model.ElectionPeriod, error) {
```

Comment:
if you know you only want to get 1 document, you should set a limit of 1
or just use the FindOne method instead of Find